module/zfs/vdev_queue.c

```diff
@@ -171,6 +171,24 @@ uint32_t zfs_vdev_rebuild_max_active = 3;
 int zfs_vdev_async_write_active_min_dirty_percent = 30;
 int zfs_vdev_async_write_active_max_dirty_percent = 60;
 
+/*
+ * To reduce effects of non-interactive I/O on interactive I/O latency
+ * the first are limited to *_min_active while there are second active,
+ * plus at least this number of I/Os after in case interactive return.
+ */
+int zfs_vdev_nia_delay = 5;
+
+/*
+ * Some HDDs tend to prioritize sequential I/O so high, that concurrent
+ * random I/O latency reaches several seconds.  On some HDDs it happens
+ * even if sequential I/Os are submitted one at a time, and so setting
+ * *_max_active to 1 does not help.  To handle this in case of scrub
+ * and other non-interactive I/O this tunable limits the number of their
+ * I/Os that can be sent until at least one interactive I/O completes
+ * without the enforced wait, making the HDD to stop the spree.
+ */
+int zfs_vdev_nia_credit = 5;
```

Comment:
It would be convenient to be able to set this value to 0 in order to disable the throttle.

Reply:
It would require additional check, while just setting it to some arbitrary high number would be effectively the same.

Reply:
It would, though some additional sanity checks might be a good idea too.  For example, preventing this from being set to a negative number.